src/structs/lepton_header.rs

```diff
@@ -68,11 +70,16 @@ pub struct LeptonHeader {
 
     pub jpeg_file_size: u32,
 
-    /// on decompression, plain-text size
-    pub plain_text_size: u32,
+    /// on decompression, uncompressed lepton header size. This is only
+    /// saved by this encoder for historical reasons. It is not used by
+    /// the decoder.
+    pub uncompressed_lepton_header_size: Option<u32>,
+
+    /// the git revision of the encoder that created this file (first 8 hex characters)
```

Comment:
Just to be sure - git revision and encoder-version used to be in C++ version and you are now implementing them in Rust, an it's all backward compatible, right?

Reply:
More questions here - I see that you support both MS + uncompressed size and flags, and the C++ git revision. So, is this change only for backward compatibility with old C++ files? Or you do not need anymore any of the MS data, and you are going back to C++ git revision data instead for new compressed files?

Reply:
Yeah I want to include the git revision information and version info for safety, but I don't want to break backward compat to the old version of the Rust library which include the flags there (I should have really put the flags elsewhere, but whatever)